BraintreeCore/src/main/java/com/braintreepayments/api/core/BraintreeHttpClient.kt

```diff
@@ -18,164 +16,99 @@ internal class BraintreeHttpClient(
 ) {
 
     /**
-     * Make a HTTP GET request to Braintree using the base url, path and authorization provided.
-     * If the path is a full url, it will be used instead of the previously provided url.
-     * @param path The path or url to request from the server via GET
-     * @param configuration configuration for the Braintree Android SDK.
-     * @param authorization
-     * @param callback [NetworkResponseCallback]
+     * Make an asynchronous Braintree authenticated HTTP request.
+     *
+     * @param request the braintree http request.
+     * @param callback See [NetworkResponseCallback].
      */
-    operator fun get(
-        path: String,
-        configuration: Configuration?,
-        authorization: Authorization?,
-        callback: NetworkResponseCallback
-    ) = get(path, configuration, authorization, HttpClient.NO_RETRY, callback)
+    @Suppress("TooGenericExceptionCaught")
+    fun sendRequest(
+        request: BraintreeHttpRequest,
+        configuration: Configuration? = null,
+        authorization: Authorization? = null,
+        callback: NetworkResponseCallback?
+    ) {
+        try {
+            val httpRequest = buildHttpRequest(request, configuration, authorization)
+            httpClient.sendRequest(httpRequest, request.retryStrategy, callback)
+        } catch (e: Exception) {
+            // forward errors
+            callback?.onResult(null, e)
```

Comment:
Take it or leave it but not sure if we want to make these more verbose as we move to Kotlin, not sure what the best practice is here:
```suggestion
        } catch (exception: Exception) {
            // forward errors
            callback?.onResult(null, exception)
```

Reply:
Yeah we could. We sometimes go with `error` I can put that in.